packages/EasyDoctrine/src/Dispatchers/DeferredEntityEventDispatcher.php

```diff
@@ -163,6 +174,7 @@ public function dispatch(): void
     public function enable(): void
```

Comment:
It was the 1st implementation of this task 🙂 
It causes a number of edge cases that we should think about and make decisions about. The current implementation is enough for Payments, so we decided to stop on it since it's easier than the 1st one.

We didn't find any edge cases for the current way. Could you please show an example of unpredictable behavior for multiple `disable($classes)` calls?

Reply:
i think that usage of `addToDisableList` + `removeFromDisableList` + enable (without args) + disable(without args) can solve all problems.

Reply:
`addToDisableList` + `removeFromDisableList` + `enable()` + `disable()` would be the same as `disable(?array $entityClasses = null)` + `enable(?array $entityClasses = null)`. And we againg face with the following case for example:
 - `disable`
 - `removeFromDisableList(A)`
 - `disable`
 - `flush`
 
So we just change the method names.

Reply:
It seems we need smth like a re-disable feature that allows us to re-disable only entities that were disabled before. Also any subsequent `enable()` call shouldn't impact high-level disabled/enabled entities.

Another thing (and it was discussed before) is that we shouldn't implement any checks in the `dispatch` method. If some events were deferred then they shouldn't be ignored even if we call `disable` in a subsequent logic. Moreover, this point is relevant even for the current logic. Let's suppose the following case based on the example above:
`<?php

class Service1
{
    public function __construct(private Service2 $service2) {}

    public function execute(): void
    {
        // ->> Changing SomeEntityA here will deffer events for it
        $this->service2->execute();
    }
}


class Service2
{
    public function __construct(private DeferredEntityEventDispatcher $deferredEntityEventDispatcher) {}

    public function execute(): void
    {
        // ...
        $this->deferredEntityEventDispatcher->disable(); -> disables dispatching even for SomeEntityA, that's wrong
    }
}`
It's quite a strange use-case but it's possible, the package doesn't forbid it.

So at this point in time, I have the following thoughts:
 - we should work with enable/disable similar to how doctrine DB-transaction works. I.e. it's always should be a paired call with isolation by level
 - we should enable/disable only events deferring but not dispatching
 - it looks that to solve all edge cases we need both of the points above
 
 It's just a draft idea and it should be discussed.